var/spack/repos/builtin/packages/gaussian-view/package.py

```diff
@@ -95,3 +103,10 @@ def setup_run_environment(self, env):
 
         # the wrappers in gv/exec setup everything just nicely
         env.prepend_path('PATH', join_path(self.prefix, 'exec'))
+
+        # except for the direct dependencies of the GUI, make sure
+        # the ones installed by Spack are being used/found
+        for dep in ['libx11', 'libxext', 'libxrender', 'libice',
+                    'libsm', 'gl', 'glu']:
+            env.append_path('LD_LIBRARY_PATH',
+                            self.spec[dep].libs.directories[0])
```

Comment:
Spack should be doing this automatically. This change shouldn't be needed.

Reply:
this is the module file without that part:

```
#%Module1.0
## Module file created by spack (https://github.com/spack/spack) on 2021-03-17 17:52:42.570109
##
## gaussian-view@6.1.6%gcc@7.5.0+gaussian-src arch=linux-opensuse_leap15-sandybridge/wagb7h3
##


module-whatis "GaussView 6 is the latest iteration of a graphical interface used with Gaussian. It aids in the creation of Gaussian input files, enables the user to run Gaussian calculations from a graphical interface without the need for using a command line instruction, and helps in the interpretation of Gaussian output."

proc ModulesHelp { } {
puts stderr "GaussView 6 is the latest iteration of a graphical interface used with"
puts stderr "Gaussian. It aids in the creation of Gaussian input files, enables the"
puts stderr "user to run Gaussian calculations from a graphical interface without the"
puts stderr "need for using a command line instruction, and helps in the"
puts stderr "interpretation of Gaussian output. Needs post-install steps to make it"
puts stderr "run! See package installation logs for details."
}

if { [ module-info mode load ] && ![ is-loaded gaussian/16-C.01-pgi-18.10-zldqaom ] } {
    module load gaussian/16-C.01-pgi-18.10-zldqaom
}

prepend-path LD_LIBRARY_PATH "/sw-ng/spack/opt/spack/linux-opensuse_leap15-sandybridge/gcc-7.5.0/gaussian-view-6.1.6-wagb7h3zo6lgshpnoblbfv6aigfut7yp/lib"
prepend-path PATH "/sw-ng/spack/opt/spack/linux-opensuse_leap15-sandybridge/gcc-7.5.0/gaussian-view-6.1.6-wagb7h3zo6lgshpnoblbfv6aigfut7yp/bin"
prepend-path CMAKE_PREFIX_PATH "/sw-ng/spack/opt/spack/linux-opensuse_leap15-sandybridge/gcc-7.5.0/gaussian-view-6.1.6-wagb7h3zo6lgshpnoblbfv6aigfut7yp/"
prepend-path XLOCALEDIR "/sw-ng/spack/opt/spack/linux-opensuse_leap15-sandybridge/gcc-7.5.0/libx11-1.7.0-vjcdmapajed4qeoqhfyah3ey2mf4b5e7/share/X11/locale"
prepend-path PYTHONPATH ""
setenv GV_DIR "/sw-ng/spack/opt/spack/linux-opensuse_leap15-sandybridge/gcc-7.5.0/gaussian-view-6.1.6-wagb7h3zo6lgshpnoblbfv6aigfut7yp"
prepend-path PATH "/sw-ng/spack/opt/spack/linux-opensuse_leap15-sandybridge/gcc-7.5.0/gaussian-view-6.1.6-wagb7h3zo6lgshpnoblbfv6aigfut7yp/exec"
```

and the error I get when trying to run `gview`:

```
/sw-ng/spack/opt/spack/linux-opensuse_leap15-sandybridge/gcc-7.5.0/gaussian-view-6.1.6-wagb7h3zo6lgshpnoblbfv6aigfut7yp/gview.exe: error while loading shared libraries: libGLU.so.1: cannot open shared object file: No such file or directory
```

again, there is no compilation or linking done.

Reply:
@adamjstewart I would like to avoid having to expose the `libX11` (and other libs) as module files just to get them loaded. Other ideas?

Reply:
I see. So the issue is that `gaussian-view` installs a pre-compiled binary, so no RPATHing happens, and the dependent modules need to be loaded for the package to work. @scheibelp is this the right approach, or should we make these packages run dependencies or something?

Reply:
Just making them run dependencies doesn't help unfortunately.
I guess the other solution would be to use the `direct` module loading mechanism, but this requires to build module files (something which is not needed if they'd be built from source).
Yet another solution might be direct ELF object manipulation, but it's still a bit unclear to me whether there must already be enough space in the ELF object to hold the additional paths or whether this can be added completely dynamically.

Btw, `vmd` suffers from the same thing. Without a working `mesa` installation on the host it will not run.

Reply:
Since I'm now using `patchelf` instead this is resolved.